pages/dao/[symbol]/proposal/components/instructions/UXDProtocol/MintWithIdentityDepository.tsx

```diff
@@ -0,0 +1,210 @@
+import {
+  UiInstruction,
+  UXDMintWithIdentityDepositoryForm,
+} from '@utils/uiTypes/proposalCreationTypes'
+import useWalletStore from 'stores/useWalletStore'
+import { NewProposalContext } from '../../../new'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+import { PublicKey, TransactionInstruction } from '@solana/web3.js'
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  Token,
+  TOKEN_PROGRAM_ID,
+} from '@solana/spl-token'
+import { findATAAddrSync } from '@utils/ataTools'
+
+async function checkInitTokenAccount(
+  account: PublicKey,
+  instructions: TransactionInstruction[],
+  connection: any,
+  mint: PublicKey,
+  owner: PublicKey,
+  feePayer: PublicKey
+) {
+  const accountInfo = await connection.current.getAccountInfo(account)
+  if (accountInfo && accountInfo.lamports > 0) {
+    return
+  }
+  instructions.push(
+    Token.createAssociatedTokenAccountInstruction(
+      ASSOCIATED_TOKEN_PROGRAM_ID, // always ASSOCIATED_TOKEN_PROGRAM_ID
+      TOKEN_PROGRAM_ID, // always TOKEN_PROGRAM_ID
+      mint, // mint
+      account, // ata
+      owner, // owner of token account
+      feePayer
+    )
+  )
+}
+const schema = yup.object().shape({
+  governedAccount: yup
+    .object()
+    .nullable()
+    .required('Governance account is required'),
+  uxdProgram: yup.string().required('UXD Program address is required'),
+  collateralAmount: yup
+    .number()
+    .moreThan(0, 'Collateral amount should be more than 0')
+    .required('Collateral Amount is required'),
+})
+
+const MintWithIdentityDepository = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const connection = useWalletStore((s) => s.connection)
+  const wallet = useWalletStore((s) => s.current)
+  const shouldBeGoverned = !!(index !== 0 && governance)
+  const [formErrors, setFormErrors] = useState({})
+  const { handleSetInstructions } = useContext(NewProposalContext)
+  const { assetAccounts } = useGovernanceAssets()
+
+  const [form, setForm] = useState<UXDMintWithIdentityDepositoryForm>({
+    governedAccount: undefined,
+    collateralAmount: 0,
+    uxdProgram: 'UXD8m9cvwk4RcSxnX2HZ9VudQCEeDH6fRnB4CAP57Dr',
+  })
+
+  const handleSetForm = ({ propertyName, value }) => {
+    setFormErrors({})
+    setForm({ ...form, [propertyName]: value })
+  }
+
+  const validateInstruction = async (): Promise<boolean> => {
+    const { isValid, validationErrors } = await isFormValid(schema, form)
+    setFormErrors(validationErrors)
+    return isValid
+  }
+
+  async function getInstruction(): Promise<UiInstruction> {
+    const isValid = await validateInstruction()
+
+    if (
+      !connection ||
+      !isValid ||
+      !form.governedAccount?.governance?.account.governedAccount ||
+      !wallet?.publicKey
+    ) {
+      return {
+        serializedInstruction: '',
+        isValid: false,
+        governance: form.governedAccount?.governance,
+      }
+    }
+    const uxdProgramId =
```

Comment:
At first glance it doesn't seems like we use `form.uxdProgram` value everywhere?

What if we use ` form.governedAccount?.governance?.account.governedAccount` for client init and controller init, but `new PublicKey(form.uxdProgram)` when creating the mint instruction?